Sources/ProjectAutomation/TargetDependency.swift

```diff
@@ -11,7 +11,7 @@ public enum TargetDependency: Equatable, Hashable, Codable {
     case framework(path: String)
     case xcframework(path: String)
     case library(path: String, publicHeaders: String, swiftModuleMap: String?)
-    case package(product: String)
+    case package(product: String, isPlugin: Bool)
```

Comment:
I think it's worth making a separate case. As a basis, you can take an analog from [SPM](https://github.com/apple/swift-package-manager/blob/0100d605e304e57b147b1172ed1442e27f9a6c52/Sources/PackageDescription/Target.swift#L230):    

```swift
/// A plug-in used in a target.
    @available(_PackageDescription, introduced: 5.5)
    public enum PluginUsage {
        /// Specifies the use of a plug-in product in a package dependency.
        ///
        /// - Parameters:
        ///   - name: The name of the plug-in target.
        ///   - package: The name of the package that defines the plug-in target.
        case plugin(name: String, package: String?)
    }
```

As an example like this:
```swift
 case plugin(name: String, package: String?)
```

Reply:
Thought about this as well, but am not entirely convinced since technically you are referencing a product within a package and having a separate `plugin` case could bring more confusion (and effort) than value.
But no strong feelings here. If core contributors think it should be a dedicated case I'm happy to change the implementations.

Reply:
👋🏼 Chiming in. Both options seem sensible. Calling it `plugin` would make it consistent with SPM's interface, but in the context of Tuist targets it's not obvious that we are referring to a Swift Package Plugin. `.package` makes it obvious, but we lose the consistency bit. What if we do something in between? We can name the case `packagePlugin`? By adding a new case we can also introduce attributes in the future that are specific to package plugins.

Reply:
ok I added `packagePlugin` to 3 different enums. see commits:
https://github.com/tuist/tuist/pull/5299/commits/d4461a245732f9f74e0cd6a861d1a1b1aabd992c
https://github.com/tuist/tuist/pull/5299/commits/c8078f7804ad057bb4d766bdbb73face3e2639ca
https://github.com/tuist/tuist/pull/5299/commits/8fc3041db2ae11f1dc47281ea2933fe54462c141